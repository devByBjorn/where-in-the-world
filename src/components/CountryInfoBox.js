import React from 'react'
import {
  HeadingCountry,
  ImgStyled,
  InfoBox,
  Item,
  LinkStyled,
  List,
  ListItem,
  ListItemSpan,
  OverLay,
  TextWrapper,
} from './countryInfoBoxStyled'
import useMediaQuery from '../hooks/useMediaQuery'

const CountryInfoBox = ({
  flag,
  name,
  population,
  region,
  capital }) => {

  const mediaQuery = useMediaQuery()

  return (
    <LinkStyled to={`/country/${name}`}>
      {mediaQuery.width > 1025 && <OverLay />}
      <InfoBox>
        <Item>
          <ImgStyled src={`${flag}`} alt={`Flag of ${name}`} />
        </Item>
        <Item>
          <TextWrapper>
            <HeadingCountry>
              {name}
            </HeadingCountry>
            <List>
              <ListItem>
                <ListItemSpan>Population: </ListItemSpan>
                {new Intl.NumberFormat().format(population)}</ListItem>
              <ListItem>
                <ListItemSpan>Region: </ListItemSpan>
                {region}</ListItem>
              <ListItem>
                <ListItemSpan>Capital: </ListItemSpan>
                {capital}</ListItem>
            </List>
          </TextWrapper>
        </Item>
      </InfoBox>
    </LinkStyled>
  )

}

export default CountryInfoBox


