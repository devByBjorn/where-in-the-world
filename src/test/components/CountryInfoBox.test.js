import React from "react"
import ReactDOM from "react-dom"
import { shallow } from 'enzyme'
import CountryInfoBox from '../../components/CountryInfoBox'

const country = {
  flag: 'https://restcountries.eu/data/afg.svg',
  name: 'Afghanistan',
  population: 27657145,
  region: 'Asia',
  capital: 'Kabul',
}

const { flag, name, population, region, capital } = country

let wrapper

beforeEach(() => {
  wrapper = shallow(
    <CountryInfoBox
      flag={flag}
      name={name}
      population={population}
      region={region}
      capital={capital}
    />)
})

test('should render <CountryInfoBox/> with data', () => {
  expect(wrapper).toMatchSnapshot()
})
