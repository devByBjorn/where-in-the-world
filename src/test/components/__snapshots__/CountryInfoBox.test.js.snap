// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <CountryInfoBox/> with data 1`] = `
<Styled(Link)
  to="/country/Afghanistan"
>
  <styled.div>
    <styled.div>
      <styled.img
        alt="Flag of Afghanistan"
        src="https://restcountries.eu/data/afg.svg"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.h4>
          Afghanistan
        </styled.h4>
        <styled.ul>
          <styled.li>
            <styled.span>
              Population: 
            </styled.span>
            27,657,145
          </styled.li>
          <styled.li>
            <styled.span>
              Region: 
            </styled.span>
            Asia
          </styled.li>
          <styled.li>
            <styled.span>
              Capital: 
            </styled.span>
            Kabul
          </styled.li>
        </styled.ul>
      </styled.div>
    </styled.div>
  </styled.div>
</Styled(Link)>
`;
